watertap/unit_models/tests/unit_test_harness.py

```diff
@@ -0,0 +1,175 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+
+from pyomo.environ import Block, assert_optimal_termination, value
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util.exceptions import InitializationError
+import idaes.core.util.scaling as iscale
+
+
+# -----------------------------------------------------------------------------
+class UnitAttributeError(AttributeError):
+    """
+    WaterTAP exception for generic attribute errors arising from unit model testing.
+    """
+
+
+class UnitValueError(ValueError):
+    """
+    WaterTAP exception for generic value errors arising from unit model testing.
+    """
+
+
+class UnitRuntimeError(RuntimeError):
+    """
+    WaterTAP exception for generic runtime errors arising from unit model testing.
+    """
+
+
+class UnitTestHarness:
+    def configure_class(self):
+        self.solver = None  # string for solver, if None use WaterTAP default
+        self.optarg = (
+            None  # dictionary for solver options, if None use WaterTAP default
+        )
+
+        self.configure()
+        blk = self.unit_model_block
+
+        # attaching objects to model to carry through in pytest frame
+        assert not hasattr(blk, "_test_objs")
+        blk._test_objs = Block()
+        blk._test_objs.solver = self.solver
+        blk._test_objs.optarg = self.optarg
+        blk._test_objs.stateblock_statistics = self.unit_statistics
+        blk._test_objs.unit_solutions = self.unit_solutions
```

Comment:
We always have access to `self`, so I don't understand the need to attach these objects to the block.